File: menuapp/lib/Model/chietkhau.dart
class ChietKhau {
  final int chietKhauId;
  final String maChietKhau;
  final int giaTriChietKhau; 
  final DateTime ngayBatDau;
  final DateTime ngayKetThuc;
  final int soLuong;

  ChietKhau({
    required this.chietKhauId,
    required this.maChietKhau,
    required this.giaTriChietKhau,
    required this.ngayBatDau,
    required this.ngayKetThuc,
    required this.soLuong,
  });

  // Tạo đối tượng từ JSON
  factory ChietKhau.fromJson(Map<String, dynamic> json) {
    return ChietKhau(
      chietKhauId: json['chietKhauId'],
      maChietKhau: json['maChietKhau'],
      giaTriChietKhau: json['giaTriChietKhau'],
      ngayBatDau: DateTime.parse(json['ngayBatDau']),
      ngayKetThuc: DateTime.parse(json['ngayKetThuc']),
      soLuong: json['soLuong'],
    );
  }

  // Chuyển đối tượng sang JSON
  Map<String, dynamic> toJson() {
    return {
      'chietKhauId': chietKhauId,
      'maChietKhau': maChietKhau,
      'giaTriChietKhau': giaTriChietKhau,
      'ngayBatDau': ngayBatDau.toIso8601String(),
      'ngayKetThuc': ngayKetThuc.toIso8601String(),
      'soLuong': soLuong,
    };
  }
}

File: ADMIN/ADMIN/Repository/Models/ChietKhau.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ADMIN.Repository.Models
{
    [Table("ChietKhau")]
    public partial class ChietKhau
    {
        [Key]
        [Column("ChietKhauId")]
        public int ChietKhauId { get; set; }

        [Column("MaChietKhau")]
        [Required(ErrorMessage = "Mã chiết khấu không được để trống.")]
        [StringLength(50)]
        public string MaChietKhau { get; set; } = null!;

        [Column("GiaTriChietKhau")]
        [Required(ErrorMessage = "Giá trị chiết khấu không được để trống.")]
        [Range(1, int.MaxValue, ErrorMessage = "Giá trị chiết khấu phải lớn hơn 0.")]
        public int GiaTriChietKhau { get; set; }
        [DisplayFormat(DataFormatString = "{0:dd/MM/yyyy}", ApplyFormatInEditMode = true)]
        [Column("NgayBatDau", TypeName = "date")]
        [Required(ErrorMessage = "Ngày bắt đầu không được để trống.")]
        public DateTime NgayBatDau { get; set; }
        [DisplayFormat(DataFormatString = "{0:dd/MM/yyyy}", ApplyFormatInEditMode = true)]
        [Column("NgayKetThuc", TypeName = "date")]
        [Required(ErrorMessage = "Ngày kết thúc không được để trống.")]

        public DateTime NgayKetThuc { get; set; }

        [Column("SoLuong")]
        [Required(ErrorMessage = "Số lượng chiết khấu không được để trống.")]
        [Range(1, int.MaxValue, ErrorMessage = "Số lượng không hợp lệ.")]
        public int Soluong { get; set; }
    }
}
